Implement special handling on QubitDevice.var for Projector observable to improve memory usage (#1368)
PennyLaneAI/pennylane: .github/CHANGELOG.md

<h3>Improvements</h3>

> * Implement special handling on QubitDevice.var for Projector observable to improve memory usage.

Reviewer: ```suggestion
* Implement special handling for measuring the variance of Projector observables to improve memory usage.
```
